### PR TITLE
PR for wishlist

### PR DESCRIPTION
**Pull request by manismk**
- Added wishlist feature
- Added private routing for private pages
- [Link](https://deploy-preview-4--jstees-react.netlify.app/)

### CODE REVIEW

File: src/components/PrivateRoute/PrivateRoute.jsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Navigate, Outlet } from "react-router-dom";
+import { useAuth } from "../../context/auth-context";
+
+export const PrivateRoute = () => {
```
Comment (shobit1337) [inline src/components/PrivateRoute/PrivateRoute.jsx:4]: This component can be refactored to be more reusable and making it a wrapper component.

Look at the example in this blog: https://dev.to/iamandrewluca/private-route-in-react-router-v6-lg5

If you have any doubt feel free to ping me on discord.

File: src/context/wishlist-context.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      (async () => {
+        try {
+          const { data, status } = await axios.get(`/api/user/wishlist`, {
+            headers: { authorization: localStorage.getItem("token") },
+          });
+          if (status === 200) {
+            setWishlist(data.wishlist);
+          }
+        } catch (e) {
+          console.log("Error in Getting Wishlist", e);
```
Comment (shobit1337) [inline src/context/wishlist-context.js]: For these kinds of messages, you can use: `console.error()`

**Review by shobit1337 — APPROVED** [review]
Suggested some good practices to follow.

Rest your code looks good to me. You've created helper methods and the code looks clean.

File: src/App.js [hunk 3]
```diff
@@ -9,6 +9,7 @@ import { Cart } from "./Pages/Cart/Cart";
 import { Profile } from "./Pages/Profile/Profile";
 import { SingleProduct } from "./Pages/SingleProduct/SingleProduct";
```
Comment (znjs) [inline src/App.js:10]: Suggestion: use barrel export if possible

File: src/App.js [hunk 4]
```diff
@@ -9,6 +9,7 @@ import { Cart } from "./Pages/Cart/Cart";
...
 import Mockman from "mockman-js";
+import { PrivateRoute } from "./components/PrivateRoute/PrivateRoute";
```
Comment (znjs) [inline src/App.js:12]: Suggestion: can use barrel export

File: src/Pages/Wishlist/Wishlist.jsx [hunk 5]
```diff
@@ -1,17 +1,27 @@
+import axios from "axios";
 import { Navbar } from "../../components/Navbar/Navbar";
 import { ProductCard } from "../../components/ProductCard/ProductCard";
+import { useWishlist } from "../../context/wishlist-context";
```
Comment (znjs) [inline src/Pages/Wishlist/Wishlist.jsx]: Suggestion: can use barrel export

File: src/components/ProductCard/components/CloseButton.jsx [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    <button
+      className="btn icon--btn icon--badge"
+      disabled={isDisabled}
+      onClick={() => {
+        if (!authData.isLoggedIn) {
+          navigate("/login");
+        }
+        if (authData.isLoggedIn) {
+          setDisabled((prev) => !prev);
+          handleDeleteWishlist(product);
```
Comment (samflab) [inline src/components/ProductCard/components/CloseButton.jsx]: use ternary instead of `if`

File: src/context/wishlist-context.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+            {
+              headers: { authorization: localStorage.getItem("token") },
+            }
+          );
+          if (status === 201) {
+            setWishlist(data.wishlist);
+          }
+        }
+      } catch (e) {
+        console.log("Error in Adding wishlist", e);
```
Comment (znjs) [inline src/context/wishlist-context.js]: Can use `console.error()` for printing error messages

**Review by samflab — APPROVED** [review]
Your code is readable and easy to understand. Also, the folder structure is clean.

File: src/context/wishlist-context.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+          `/api/user/wishlist/${product._id}`,
+          {
+            headers: { authorization: localStorage.getItem("token") },
+          }
+        );
+        if (status === 200) {
+          setWishlist(data.wishlist);
+        }
+      } catch (e) {
+        console.log("Error in Deleting wishlist", e);
```
Comment (znjs) [inline src/context/wishlist-context.js]: Can use `console.error()` for printing error messages

File: src/index.js [hunk 9]
```diff
@@ -6,6 +6,7 @@ import { makeServer } from "./server";
 import { BrowserRouter } from "react-router-dom";
 import { ProductDataProvider } from "./context/product-data-context";
 import { AuthProvider } from "./context/auth-context";
+import { WishlistProvider } from "./context/wishlist-context";
```
Comment (znjs) [inline src/index.js]: Can use barrel export so that all Providers can be imported directly from `./context`

**Review by znjs — APPROVED** [review]

**Comment by samflab** [comment]
Just a UI suggestion, reduce the box shadow of the navbar.
Add a right border to the filter component so that it differentiates from the product page. 
add some margin to the checkbox/radio and the label.
change the background-color of the page, don't keep it all yellow, change it to white or off-white

image

**Review by Vikas2426 — APPROVED** [review]

File: src/components/PrivateRoute/PrivateRoute.jsx (hunk 1 above)
Comment (manismk) [reply src/components/PrivateRoute/PrivateRoute.jsx:4]: Thanks @shobit1337 will read about it and implement it.

File: src/context/wishlist-context.js (hunk 2 above)
Comment (manismk) [reply src/context/wishlist-context.js]: Done

File: src/components/ProductCard/components/CloseButton.jsx (hunk 6 above)
Comment (manismk) [reply src/components/ProductCard/components/CloseButton.jsx]: Thanks @samflab for detailed review made most of the change

File: src/index.js (hunk 9 above)
Comment (manismk) [reply src/index.js]: Thanks @znjs  done most of the barrel export and console error

**Pull request merged**